### PR TITLE
2.7 tree level

### PR DESCRIPTION
**Pull request by ADmad**
Backported from 3.0

### CODE REVIEW

**Comment by markstory** [comment]
Looks good to me.

File: lib/Cake/Model/Behavior/TreeBehavior.php [hunk 1]
```diff
@@ -97,11 +97,48 @@ public function afterSave(Model $Model, $created, $options = array()) {
...
+ * @return void
+ */
+	protected function _setChildrenLevel($Model, $id) {
+		$settings = $Model->Behaviors->Tree->settings[$Model->alias];
+		$primaryKey = $Model->primaryKey;
+		$depths = array($id => (int)$Model->data[$Model->alias][$settings['level']]);
+
+		$children = $Model->children(
+			$id,
+			false,
```
Comment (AD7six) [inline lib/Cake/Model/Behavior/TreeBehavior.php]: gotta love that method signature.

Not.

**Comment by AD7six** [comment]
:+1:

File: lib/Cake/Model/Behavior/TreeBehavior.php (hunk 1 above)
Comment (markstory) [inline lib/Cake/Model/Behavior/TreeBehavior.php]: I am glad we stopped doing stuff like that for the most part.

Comment (ADmad) [inline lib/Cake/Model/Behavior/TreeBehavior.php]: I had to keep the api open side by side to ensure i got the order of params right :stuck_out_tongue:

**Pull request merged**